### PR TITLE
(BKR-327) Add support for EL7 to epel_info_for and add_el_extras method

### PR DESCRIPTION
**Pull request by bkero**
Some new code to hopefully pass tests. Adds support for epel7 in epel_info_for and add_el_extras, as wel as adding a test for el7 url retrieval.

### CODE REVIEW

**Comment by anodelman** [comment]
ok to test

**Comment by bkero** [comment]
@anodelman can you please paste some outout from the failing build? I don't have access to that jenkins host.

**Comment by anodelman** [comment]
The failure looks transient - I'll rerun.

**Comment by anodelman** [comment]
retest this please

File: lib/beaker/host_prebuilt_steps.rb [hunk 1]
```diff
@@ -170,16 +170,23 @@ def epel_info_for host, opts
       end
 
       version = host['platform'].version
-      if version == '6'
-        url = "#{host[:epel_url] || opts[:epel_url]}/#{version}"
+      url = "#{host[:epel_url] || opts[:epel_url]}/#{version}"
+      if version == '7'
+        if opts[:epel_7_arch] == 'i386'
+          raise "epel-7 does not provide packages for i386"
```
Comment (anodelman) [inline lib/beaker/host_prebuilt_steps.rb]: Just a nit to raise this as an ArgumentError.

File: lib/beaker/host_prebuilt_steps.rb [hunk 2]
```diff
@@ -170,16 +170,23 @@ def epel_info_for host, opts
...
+        arch = opts[:epel_7_arch] || 'x86_64'
+      elsif version == '6'
         pkg = host[:epel_pkg] || opts[:epel_6_pkg]
+        arch = host[:epel_arch] || opts[:epel_6_arch] || 'i386'
       elsif version == '5'
-        url = "#{host[:epel_url] || opts[:epel_url]}/#{version}"
         pkg = host[:epel_pkg] || opts[:epel_5_pkg]
+        arch = host[:epel_arch] || opts[:epel_5_arch] || 'i386'
       else
         raise "epel_info_for does not support el version #{version}, on #{host.name}"
```
Comment (anodelman) [inline lib/beaker/host_prebuilt_steps.rb]: While you're in here, let's get this updated to an ArgumentError as well.  :)

**Comment by anodelman** [comment]
Just a couple of nits and then ready to go.  \o/

**Comment by anodelman** [comment]
spec failure:
```
Failures:

  1) Beaker epel_info_for! raises an error on non el-5/6 host
     Failure/Error: expect{ subject.epel_info_for( host, options )}.to raise_error(RuntimeError, /epel_info_for does not support el version/)
       expected RuntimeError with message matching /epel_info_for does not support el version/, got #<ArgumentError: epel_info_for does not support el version 4, on testhost> with backtrace:
         # ./lib/beaker/host_prebuilt_steps.rb:187:in `epel_info_for'
         # ./spec/beaker/host_prebuilt_steps_spec.rb:179:in `block (4 levels) in <top (required)>'
         # ./spec/beaker/host_prebuilt_steps_spec.rb:179:in `block (3 levels) in <top (required)>'
     # ./spec/beaker/host_prebuilt_steps_spec.rb:179:in `block (3 levels) in <top (required)>'

Finished in 28.54 seconds (files took 4.76 seconds to load)
1049 examples, 1 failure
```

**Pull request merged**